docs(service-worker): various small fixes/improvements in SW docs (#37555)
angular/angular: aio/content/guide/service-worker-communications.md

### Checking for updates

It's possible to ask the service worker to check if any updates have been deployed to the server.

> The service worker checks for updates during initialization and on each navigation request (i.e. when the user navigates from a different address to your app).

Reviewer: ```suggestion
The service worker checks for updates during initialization and on each navigation request&mdash;that is, when the user navigates from a different address to your app.
```
Or any other way you'd like to say it without parentheses or "e.g." (no latin).

However, you might choose to manually check for updates if you have a site that changes frequently or want updates to happen on a schedule.

Do this with the `checkForUpdate()` method:

<code-example path="service-worker-getting-started/src/app/check-for-update.service.ts" header="check-for-update.service.ts"></code-example>

This method returns a `Promise` which indicates that the update check has completed successfully, though it does not indicate whether an update was discovered as a result of the check. Even if one is found, the service worker must still successfully download the changed files, which can fail. If successful, the `available` event will indicate availability of a new version of the app.

<div class="alert is-important">

> In order to avoid negatively affecting the initial rendering of the page, `ServiceWorkerModule` will by default wait for up to 30 seconds for the app to stabilize, before registering the ServiceWorker script.

Reviewer: ```suggestion
In order to avoid negatively affecting the initial rendering of the page, `ServiceWorkerModule` waits for up to 30 seconds by default for the app to stabilize, before registering the ServiceWorker script.
```

> Constantly polling for updates, e.g. with [setInterval()](https://developer.mozilla.org/en-US/docs/Web/API/WindowOrWorkerGlobalScope/setInterval) or RxJS' [interval()](https://rxjs.dev/api/index/function/interval), will prevent the app from stabilizing and the ServiceWorker script will not be registered with the browser until the 30 seconds upper limit is reached.

Reviewer: ```suggestion
Constantly polling for updates, for example, with [setInterval()](https://developer.mozilla.org/en-US/docs/Web/API/WindowOrWorkerGlobalScope/setInterval) or RxJS' [interval()](https://rxjs.dev/api/index/function/interval), will prevent the app from stabilizing and the ServiceWorker script will not be registered with the browser until the 30 seconds upper limit is reached.
```

Note that this is true for any kind of polling done by your application.
Check the {@link ApplicationRef#isStable isStable} documentation for more information.

> You can avoid that delay by waiting for the app to stabilize first, before starting to poll for updates (as shown in the example above).

Reviewer: ```suggestion
You can avoid that delay by waiting for the app to stabilize first, before starting to poll for updates , as in the example above.
```
